File: backend/app/models/rate.py
from pydantic import BaseModel
from enum import Enum
from datetime import datetime


class Currency(str, Enum):
    USD = "USD"
    EUR = "EUR"


class InstitutionType(str, Enum):
    bank = "bank"
    fintech = "fintech"
    broker = "broker"
    exchange_house = "exchange_house"


class Rate(BaseModel):
    institution: str
    type: InstitutionType
    currency: Currency
    buy_rate: float
    sell_rate: float
    spread_pct: float
    amount_received: float  # quanto o usuário recebe na moeda estrangeira
    url: str
    updated_at: datetime


class RatesResponse(BaseModel):
    currency: Currency
    amount_brl: float
    rates: list[Rate]
    fetched_at: datetime

File: frontend/src/types/rate.ts
export type Currency = 'USD' | 'EUR'
export type InstitutionType = 'bank' | 'fintech' | 'broker' | 'exchange_house'

export interface Rate {
  institution: string
  type: InstitutionType
  currency: Currency
  buy_rate: number
  sell_rate: number
  spread_pct: number
  amount_received: number
  url: string
  updated_at: string
}

export interface RatesResponse {
  currency: Currency
  amount_brl: number
  rates: Rate[]
  fetched_at: string
}
